kernel/kernel-api/src/main/java/io/delta/kernel/ConcurrentWriteException.java

```diff
@@ -13,11 +13,22 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package io.delta.kernel;
 
-package io.delta.kernel.internal.lang;
+import java.util.ConcurrentModificationException;
 
-import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.annotation.Evolving;
 
-public interface CloseableIterable<T> {
-    CloseableIterator<T> iterator();
+/**
+ * Thrown when a concurrent transaction has written data after the current transaction read the
```

Comment:
This isn't the only condition right? Blind append for example might be due to reaching max number of retries despite not ever reading the table.